### PR TITLE
Give cargo-dist a try-out

### PR DESCRIPTION
**Pull request by darach**
Signed-off-by: Darach Ennis <darach@gmail.com>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+# will hard limit this to 3 tags per commit, as it will assume more tags is a
+# mistake.
+#
+# If there's a prerelease-style suffix to the version, then the release(s)
+# will be marked as a prerelease.
+on:
+  push:
+    tags:
+      - '**[0-9]+.[0-9]+.[0-9]+*'
+  pull_request:
```
Comment (Licenser) [inline .github/workflows/release.yml:44]: I don't think we want to run this on every pull request

Comment (darach) [reply .github/workflows/release.yml:44]: It skips for PRs when its configured for `plan` mode and doesn't generate artefacts. There is another mode that always builds release artefacts but I disabled it. The default mode is `plan`. If you look at our CI output for this PR you'll see plan was triggered and it skipped the release distribution process.

Comment (darach) [reply .github/workflows/release.yml:44]: Here's the reference in the docs:

https://opensource.axo.dev/cargo-dist/book/introduction.html#distributing

My read of that is that until we push a tag it does nothing. Our release process adds a tag so it should only fire on release for us with no other configuration.

Comment (Licenser) [reply .github/workflows/release.yml:44]: Interesting, that's complex but lets give it a shot 👍

**Review by Licenser — APPROVED** [review]

**Pull request merged**